src/core/shape_inference/include/istft_shape_inference.hpp

```diff
@@ -0,0 +1,128 @@
+                           frame_size_shape.rank().compatible(0),
+                           "The shape of frame_size must be a scalar.");
+    NODE_SHAPE_INFER_CHECK(op,
+                           input_shapes,
+                           frame_step_shape.rank().compatible(0),
+                           "The shape of frame_step must be a scalar.");
+
+    const auto frame_size = get_input_const_data_as<TRShape, int64_t>(op, 2, ta);
+    const auto frame_step = get_input_const_data_as<TRShape, int64_t>(op, 3, ta);
+
+    if (frame_size) {
+        const auto& frame_size_val = (*frame_size)[0];
+        NODE_SHAPE_INFER_CHECK(op,
+                               input_shapes,
+                               0 < frame_size_val,
+                               "Provided frame size is ",
+                               frame_size_val,
+                               " but must be greater than zero.");
+        const bool is_win_shape_correct =
+            window_shape.is_dynamic() || (TDimVal{0} < window_shape[0].get_length() &&
+                                          window_shape[0].get_length() <= static_cast<TDimVal>(frame_size_val));
+
+        NODE_SHAPE_INFER_CHECK(op,
+                               input_shapes,
+                               is_win_shape_correct,
+                               "Window input dimension must be in range [1, ",
+                               frame_size_val,
+                               "].");
+    }
+
+    if (frame_step) {
+        const auto& frame_step_val = (*frame_step)[0];
+        NODE_SHAPE_INFER_CHECK(op,
+                               input_shapes,
+                               0 < frame_step_val,
+                               "Provided frame step is ",
+                               frame_step_val,
+                               " but must be greater than zero.");
+    }
+
+    // For the input with dynamic rank, output shape is also fully dynamic
+    if (data_shape_rank.is_dynamic()) {
+        return {data_shape};
+    }
+    const auto is_data_3D = data_shape.size() == 3;
+
+    std::vector<TRShape> output_shapes;
+    if (inputs_count == 5) {
+        const auto& length_shape = input_shapes[4];
+        const bool has_len_valid_shape =
+            length_shape.rank().is_dynamic() ||
+            (length_shape.size() == 0 || (length_shape.size() == 1 && length_shape[0].compatible(1)));
+        NODE_SHAPE_INFER_CHECK(op,
+                               input_shapes,
+                               has_len_valid_shape,
+                               "The shape of 'signal_length' input must be a scalar or single element 1D tensor.");
+
+        const auto sig_len_in = get_input_const_data_as_shape<TRShape>(op, 4, ta);
+        if (sig_len_in) {  // Set desired length of the signal dimension, if provided
+            output_shapes.emplace_back(TRShape{(*sig_len_in)[0]});
+        } else {
+            output_shapes.emplace_back(TRShape{TDim(ov::util::dim::inf_bound)});
```

Comment:
This could be added when output shape created and the dimension can be updated replaced depends on condition

Reply:
Done